src/main/java/com/carrotcreative/cream/util/HashingUtil.java

```diff
@@ -0,0 +1,52 @@
+package com.carrotcreative.cream.util;
+
+import java.io.UnsupportedEncodingException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+import android.support.annotation.NonNull;
+
+public class HashingUtil
+{
+    private static final int CACHE_SIZE = 50;
+    private static LruCache<String, String> mCache = new LruCache<>(CACHE_SIZE);
+
+    private HashingUtil() {}
+
+    /**
+     *
+     * @param secret The string that needs to be hashed.
+     * @return The hashed value.
+     */
+    public static String getHash(@NonNull String secret)
+    {
+        String hashedValue = mCache.get(secret);
+        if(hashedValue == null)
+        {
+            hashedValue = generateHash(secret);
+            mCache.put(secret, hashedValue);
+        }
+
+        return hashedValue;
+    }
+
+    private static String generateHash(String secret)
+    {
+        String hashedString = null;
+        try
+        {
+            MessageDigest messageDigest = MessageDigest.getInstance("SHA-256");
+            messageDigest.update(secret.getBytes("UTF-8"));
+            hashedString = new String(messageDigest.digest());
```

Comment:
Can you log this for me and show me some example output?

I'm pretty confident it's 64 characters of output, but I just want to verify...  127 chars is our max length for the hash + delimiter + epoch + extension.